x-pack/plugins/secops/server/lib/index_fields/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexField, IndexType } from '../../../common/graphql/types';
+import { FrameworkRequest } from '../framework';
+import { Sources } from '../sources';
+import { FieldsAdapter } from './types';
+export { ElasticsearchIndexFieldAdapter } from './elasticsearch_adapter';
+
+export class IndexFields implements FieldsAdapter {
+  private adapter: FieldsAdapter;
+  private sources: Sources;
+
+  constructor(adapter: FieldsAdapter, sources: Sources) {
+    this.adapter = adapter;
+    this.sources = sources;
+  }
```

Comment:
Looking at your style above for `elasticsearch_adapter.ts`

Wouldn't this be tighter types for you to use and more consistent?

```ts
constructor(private readonly adapter: FieldsAdapter, private readonly sources: Sources) { }
```

Kind of follows the constructor injection, set it only once pattern.

Reply:
I almost did it, this way